### DISCUSSION TITLE
App host running projects and docker images

### DISCUSSION THREAD

**Discussion by allensiho** [microsoft/aspire#6567] [Q&A]
I am running an Aspire App host that points to the solutions projects and a sql server container

Because this solution does not contain all the projects,  
I have to run projects outside the solution using the built docker image that runs on kubernetes 

The problem is I can get the projects to talk to the local mssql server running but not the docker container image

Is there a way to do this with aspire or what is recommended?

Thanks

**Comment by davidfowl** [comment]
Is the docker image configured to talk to the same sql sever that you're running in a container?

**Comment by allensiho** [comment]
I pass in environment variables

I have tried a few things like 
* making my docker desktop use local network
* passing in host.docker.internal gateway 
The projects can talk to mssql on localhost,1433

**Comment by allensiho** [comment]
Yes I believe docker image us running as container

**Comment by davidfowl** [comment]
I need a diagram or app host code, there's a bunch of moving parts that are hard to understand from the text.

**Comment by allensiho** [comment]
Some sample code


 var databaseSettings = builder.Configuration.GetSection(nameof(Database)).Get<Database>()!;
 
 var db = builder.AddSqlServer("mssql", password, 1433)
     //.WithBindMount("VolumeMount.AppHost-sql-data", "/var/opt/mssql")
     .WithDataVolume()
     .WithBindMount("./SqlServer", "/usr/config")
     // Mount the SQL scripts directory into the container so that the init scripts run.
     .WithBindMount("./init", "/docker-entrypoint-initdb.d")
     // Run the custom entrypoint script on startup.
     .WithEntrypoint("/usr/config/entrypoint.sh")
     .WithSqlHealthCheck();
 
 var aservice = builder.AddProject<Projects.XXXXXXXXXX>("XXXXX")
     .WithReference(rabbitmq)
     .WithReference(db)
     .WithReference(kafka)
     .WithMassTransitSettings(massTransitSettings)
     .WithDatabaseSettings(databaseSettings)
     .WithExternalHttpEndpoints()
     .WaitFor(kafka);
 
 var dockerImageService = builder.AddContainer("docker-scheduler",
         "ghcr.io/some-image", "7.1.1")
     .WithReference(rabbitmq)
     .WithReference(db)
     .WithReference(kafka)
     .WithDatabaseSettings(databaseSettings)
     .WithEnvironment("Database__ConnectionString", "Server=host.docker.internal,1433;User ID=sa;Password=XXXX;Database=XXXXXX;TrustServerCertificate=true")
     .WithContainerRuntimeArgs("--network=host")
     .WaitFor(kafka)
     .WaitFor(db);

**Reply by davidfowl** [reply]
Are you running docker desktop on windows?

**Reply by davidfowl** [reply]
I'm not sure about --network=host here.

cc @danegsta

**Reply by danegsta** [reply]
`—network=host` is going to break a lot of Aspire’s service discovery assumptions. There’s no way it could work unless we special cased it to fall back to something closer to executable service discovery behavior. It also complicates things more than it simplifies on Windows, where it really means the WSL network, not the Windows side of the network stack, so talking between containers and host processes is still not guaranteed to just work. 

@allensiho is there a specific reason you’re running that container with host networking?

**Comment by allensiho** [comment]
I cannot get my dockerImageService to talk to Sql Container

Thanks

**Comment by allensiho** [comment]
No on MAC

**Reply by davidfowl** [reply]
Docker desktop on Mac?

**Comment by allensiho** [comment]
Yes

**Comment by allensiho** [comment]
You think it should be possible for the docker container to talk to mssql?

Thanks

**Comment by allensiho** [comment]
Hi David

Do have possible ideas for this?

Maybe somehow you can setup aspire so that docker images and projects when
launched will run under the one network or something

Thanks

**Comment by stefandenchev** [comment]
I also have a similar issue, can someone advise?
